src/main/java/bio/terra/stairway/FlightDao.java

```diff
@@ -629,6 +652,65 @@ FlightContext resume(String stairwayId, String flightId)
     return null;
   }
 
+  FlightContext makeFlightContextById(String flightId)
+      throws DatabaseOperationException, InterruptedException {
+    final String sqlGetFlight =
+        "SELECT class_name, debug_info, status "
+            + " FROM "
+            + FLIGHT_TABLE
+            + " WHERE flightid = :flightId";
+
+    for (int retry = 0; retry < SERIALIZATION_RETRIES; retry++) {
+      try (Connection connection = dataSource.getConnection();
+          NamedParameterPreparedStatement getFlightStatement =
+              new NamedParameterPreparedStatement(connection, sqlGetFlight)) {
+
+        startTransaction(connection);
+
+        getFlightStatement.setString("flightId", flightId);
+        FlightContext flightContext = null;
+        try (ResultSet rs = getFlightStatement.getPreparedStatement().executeQuery()) {
+          if (rs.next()) {
+            flightContext = makeFlightContext(connection, flightId, rs);
+          }
+        }
+        commitTransaction(connection);
+        return flightContext;
+      } catch (SQLException ex) {
+        if (!retrySqlException(ex)) {
+          handleSqlException("Failed to get flight", ex);
+          return null;
+        }
+      }
+      retryWait("makeFlight", flightId);
+    }
+    return null;
+  }
+
+  // The results set must be positioned at a row from the flight table,
+  // and the select list must include debug_info, class_name, and status.
+  private FlightContext makeFlightContext(Connection connection, String flightId, ResultSet rs)
+      throws InterruptedException, DatabaseOperationException, SQLException {
+    List<FlightInput> inputList = retrieveInputParameters(connection, flightId);
+    FlightMap inputParameters = new FlightMap(inputList);
+    FlightDebugInfo debugInfo = null;
```

Comment:
I think a better idiom here is to not initialize and declare the variable as `final`. That way it must be initialized and you don't need to come up with a default value.
```suggestion
    final FlightDebugInfo debugInfo;
```

Reply:
To maintain the existing Flight constructor, we used a setter to populate the debugInfo when it is being used.
So I think it has to be not `final`

Reply:
Sorry I wasn't clear; I mean that the local variable (not the `FlightContext` field) can be declared final. Once you do that, you no longer need the `= null` initializer. Nothing else in this method would change.